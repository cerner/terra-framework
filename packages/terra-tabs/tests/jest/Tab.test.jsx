import React from 'react';
import { v4 as uuidv4 } from 'uuid';
import Tab from '../../src/common-tabs/_Tab';

let mockSpyUuid;

beforeAll(() => {
  mockSpyUuid = jest.spyOn(uuidv4, 'v4').mockReturnValue('00000000-0000-0000-0000-000000000000');
});

afterAll(() => {
  mockSpyUuid.mockRestore();
});

describe('Tab', () => {
  it('should render a common tab as disabled when indicated', () => {
    const wrapper = shallowWithIntl(<Tab label="Label" isDisabled variant="framework" />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a common tab with custom props', () => {
    const wrapper = shallowWithIntl(<Tab label="Default" className="customClass" variant="framework" />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a closable tab', () => {
    const wrapper = shallowWithIntl(<Tab label="Default" className="customClass" variant="framework" isClosable />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('correctly applies the theme context className', () => {
    jest.spyOn(React, 'useContext')
      .mockReturnValue({
        className: 'orion-fusion-theme',
      });
    const wrapper = shallowWithIntl(<Tab label="Default" className="customClass" variant="framework" />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
