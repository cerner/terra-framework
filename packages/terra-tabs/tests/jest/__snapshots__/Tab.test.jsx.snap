// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab correctly applies the theme context className 1`] = `
<div
  aria-disabled={false}
  aria-label="Default"
  aria-selected={false}
  className="tab is-text-only orion-fusion-theme"
  data-focus-styles-enabled={true}
  data-terra-tabs-show-focus-styles={true}
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="tab"
  style={
    Object {
      "zIndex": undefined,
    }
  }
  tabIndex={-1}
  title="Default"
>
  <div
    className="inner"
  >
    <VisuallyHiddenText
      aria-hidden={true}
      id="terra-tab-delete-pane-response-5ffd6120-9e47-4631-97ee-8e1d7615f534"
      text="Terra.tabs.hint.removable"
    />
    <span
      className="label"
    >
      Default
    </span>
  </div>
</div>
`;

exports[`Tab should render a common tab as disabled when indicated 1`] = `
<div
  aria-disabled={true}
  aria-label="Label"
  aria-selected={false}
  className="tab is-disabled is-text-only"
  data-focus-styles-enabled={false}
  data-terra-tabs-show-focus-styles={true}
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="tab"
  style={
    Object {
      "zIndex": undefined,
    }
  }
  tabIndex={-1}
  title="Label"
>
  <div
    className="inner"
  >
    <VisuallyHiddenText
      aria-hidden={true}
      id="terra-tab-delete-pane-response-e6540345-0d3b-4f3d-9cd5-5f70cb19fdb1"
      text="Terra.tabs.hint.removable"
    />
    <span
      className="label"
    >
      Label
    </span>
  </div>
</div>
`;

exports[`Tab should render a common tab with custom props 1`] = `
<div
  aria-disabled={false}
  aria-label="Default"
  aria-selected={false}
  className="tab is-text-only"
  data-focus-styles-enabled={true}
  data-terra-tabs-show-focus-styles={true}
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="tab"
  style={
    Object {
      "zIndex": undefined,
    }
  }
  tabIndex={-1}
  title="Default"
>
  <div
    className="inner"
  >
    <VisuallyHiddenText
      aria-hidden={true}
      id="terra-tab-delete-pane-response-780d285e-9183-4c0e-a9be-74e3dea75f38"
      text="Terra.tabs.hint.removable"
    />
    <span
      className="label"
    >
      Default
    </span>
  </div>
</div>
`;
